### PR TITLE
Allow non-websocket enabled exchanges

### PR DESCRIPTION
**Pull request by lrascao**
Instead of running an infinite loop of handling websocket messages block forever on a channel.

### CODE REVIEW

File: keep.go [hunk 1]
```diff
@@ -176,6 +191,24 @@ func (bot *Keep) Run(ctx context.Context) {
...
 }
 
+func Loop(ctx context.Context, k *Keep, e exchange.IBotExchange, s Strategy) error {
+	// If this exchange doesn't support websockets we still need to
+	// keep running
+	if !e.IsWebsocketEnabled() {
+		gctlog := GCTLog{nil}
+		gctlog.Warnf(gctlog.ExchangeSys, "%s: no websocket support", e.GetName())
+
+		// create a channel and just block forever waiting for something
```
Comment (ydm) [inline keep.go:201]: i think it's better to wait on `<-ctx.Done()` and then return

Comment (ydm) [reply keep.go:201]: also why not just returning. i don't like leaving a blocked goroutine anyways

**Review by ydm — CHANGES_REQUESTED** [review]

File: keep.go (hunk 1 above)
Comment (lrascao) [reply keep.go:201]: that's what `Stream` currently does when there's no websocket activity

File: keep.go [hunk 2]
```diff
@@ -176,6 +191,21 @@ func (bot *Keep) Run(ctx context.Context) {
...
 }
 
+func Loop(ctx context.Context, k *Keep, e exchange.IBotExchange, s Strategy) error {
+	// If this exchange doesn't support websockets we still need to
+	// keep running
+	if !e.IsWebsocketEnabled() {
+		gctlog := GCTLog{nil}
+		gctlog.Warnf(gctlog.ExchangeSys, "%s: no websocket support", e.GetName())
+
+		<-ctx.Done()
```
Comment (MarkDzulko) [inline keep.go:201]: should this not `return nil `here?

Comment (lrascao) [reply keep.go:201]: yeah, i guess it makes sense for symmetry, on both cases it's never reached

**Review by MarkDzulko — APPROVED** [review]

**Pull request merged**